kong/hybrid/message.lua

```diff
@@ -0,0 +1,121 @@
+local _M = {}
+
+
+local bit = require("bit")
+local tablepool = require("tablepool")
+
+
+local band, bor = bit.band, bit.bor
+local lshift, rshift = bit.lshift, bit.rshift
+local string_char, string_byte = string.char, string.byte
+
+
+local POOL_NAME = "hybrid_messages"
+local _MT = { __index = _M, }
+
+
+--- convert a unsigned 32bit integer to network byte order
+local function uint32_to_bytes(num)
+  if num < 0 or num > 4294967295 then
+    error("number " .. tostring(num) .. " out of range", 2)
+  end
+
+  return string_char(band(rshift(num, 24), 0xFF),
+                     band(rshift(num, 16), 0xFF),
+                     band(rshift(num, 8), 0xFF),
+                     band(num, 0xFF))
+end
+
+
+local function bytes_to_uint32(str)
+  assert(#str == 4)
+
+  local b1, b2, b3, b4 = string_byte(str, 1, 4)
+
+  return bor(lshift(b1, 24),
+             lshift(b2, 16),
+             lshift(b3, 8),
+             b4)
+end
+
+
+function _M.new(src, dest, topic, message)
+  local self = tablepool.fetch(POOL_NAME, 0, 4)
+
+  assert(dest, "dest is required")
+  assert(topic, "topic is required")
+  assert(message, "message is required")
+
+  self.src = src
```

Comment:
since src, dest, and topic's length fit in a char, we should check their length
are shorter than 256bytes